### ISSUE TITLE
Uncaught exception: Cannot set property 'as' of undefined

### ISSUE THREAD

**Issue by jwolski** [uber/tchannel#875]
Received uncaught exception from tchannel while staging a new build on a standalone 1k-node Ringpop cluster.

I'm running tchannel 2.0.5.

```
"stack":"TypeError: Cannot set property 'as' of undefined
    at handleThriftResponse (node_modules/tchannel/as/thrift.js:164:28)
    at onCallback (node_modules/ringpop/server/index.js:173:9)
    at handlePing (node_modules/ringpop/server/ping-handler.js:36:5)
    at handleIt (node_modules/ringpop/server/index.js:110:9)
    at TChannelEndpointHandler.handleThriftRequest (node_modules/tchannel/as/thrift.js:122:9)
    at onArg23 (node_modules/tchannel/request-handler.js:57:34)
    at compatCall (node_modules/tchannel/streaming_in_request.js:76:9)
    at zalgoSafe (node_modules/dezalgo/dezalgo.js:20:10)
    at done (node_modules/tchannel/node_modules/run-parallel/index.js:18:15)
    at signalReady (node_modules/tchannel/node_modules/ready-signal/index.js:21:26)
    at StreamArg.finish node_modules/tchannel/argstream.js:263:9)
    at StreamArg.emit (events.js:117:20)
    at node_modules/tchannel/node_modules/readable-stream/lib/_stream_readable.js:965:16
    at process._tickDomainCallback (node.js:463:13)”,
"message":"Cannot set property 'as' of undefined"
```

**Comment by Raynos** [comment]
Ringpop master?

**Comment by jwolski** [comment]
`thrift` branch.

**Comment by Raynos** [comment]
This is probably something to do with streaming and large bodies.

Will investigate.

**Comment by jwolski** [comment]
You're likely right. The problem didn't arise until > 50% of the nodes had been upgraded to the new version

**Comment by Raynos** [comment]
```diff
$ git diff
diff --git a/node/request-handler.js b/node/request-handler.js
index 48d3e11..e522f7d 100644
--- a/node/request-handler.js
+++ b/node/request-handler.js
@@ -54,7 +54,8 @@ RequestCallbackHandler.prototype.handleRequest = function handleRequest(req, bui
     var res;
     if (req.streamed) {
         req.withArg23(function onArg23(err, arg2, arg3) {
-            return self.callback.call(self.thisp, req, buildResponse, arg2, arg3);
+            res = buildResponse({streamed: false});
+            return self.callback.call(self.thisp, req, res, arg2, arg3);
         });
     } else {
         res = buildResponse({streamed: false});
```

Should fix your issue; PR soon.

**Comment by Raynos** [comment]
Fixed by #877

**Comment by jwolski** [comment]
:+1:
